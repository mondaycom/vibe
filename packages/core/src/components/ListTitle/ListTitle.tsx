import cx from "classnames";
import React, { forwardRef, useRef } from "react";
import useMergeRef from "../../hooks/useMergeRef";
import VibeComponentProps from "../../types/VibeComponentProps";
import Text from "../Text/Text";
import styles from "./ListTitle.module.scss";

export interface ListTitleProps extends VibeComponentProps {
  /**
   * The tab index for keyboard navigation.
   */
  tabIndex?: number;
  /**
   * The title text.
   */
  children?: string;
}

const ListTitle: React.FC<ListTitleProps> = forwardRef(
  ({ className, id, children, tabIndex, "data-testid": dataTestId }, ref) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);

    return (
      <Text
        type="text1"
        weight="medium"
        data-testid={dataTestId || id}
        aria-level={3}
        tabIndex={tabIndex}
        role="heading"
        ref={mergedRef}
        className={cx(styles.listTitle, className)}
        id={id}
      >
        {children}
      </Text>
    );
  }
);

Object.assign(ListTitle, {
  // Used by VirtualizedListItems
  displayName: "ListTitle"
});

export default ListTitle;
