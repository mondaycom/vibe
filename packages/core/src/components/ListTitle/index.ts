export { default as ListTitle, type ListTitleProps } from "./ListTitle";
