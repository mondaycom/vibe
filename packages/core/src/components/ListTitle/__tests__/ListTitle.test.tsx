import { it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import ListTitle from "../ListTitle";

it("renders correctly with empty props", () => {
  const tree = renderer.create(<ListTitle data-testid="list-title">Title</ListTitle>).toJSON();
  expect(tree).toMatchSnapshot();
});
