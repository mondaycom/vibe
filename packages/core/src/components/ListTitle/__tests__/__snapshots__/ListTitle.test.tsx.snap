// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with empty props 1`] = `
<div
  aria-level={3}
  className="typography primary start singleLineEllipsis text text1Medium listTitle"
  data-testid="list-title"
  role="heading"
>
  Title
</div>
`;
